### PR TITLE
Geometry in parent

### PR DESCRIPTION
**Pull request by hiptmair**
Implementation and testing of `GeometryInParent()` method of `MeshHierarchy`.

### CODE REVIEW

File: lib/lf/refinement/test/refinement_test_utils.cc [hunk 1]
```diff
@@ -11,6 +11,102 @@
...
+      const lf::geometry::Geometry *geo_parent = fp->Geometry();
+      // Check matching of dimensions
+      EXPECT_EQ(geo_in_parent->DimGlobal(), geo_parent->DimLocal())
+          << "Mismatch: geo_child->DimGlobal() = " << geo_child->DimGlobal()
+          << " <-> geo_parent->DimLocal() = " << geo_parent->DimLocal();
+      // The following test is valid only for affine entities
+      if (geo_parent->isAffine()) {
+        // Test point in child reference coordinates
+        const Eigen::VectorXd testpoint =
+            Eigen::VectorXd::Constant(geo_child->DimLocal(), 1.0 / 3.0);
```
Comment (craffael) [inline lib/lf/refinement/test/refinement_test_utils.cc]: An alternative idea would be to use the nodes of a quadrature rule for the respective RefEl

File: lib/lf/refinement/test/refinement_test_utils.cc [hunk 2]
```diff
@@ -11,6 +11,102 @@
...
+        // Map test point through the geometry object for the child entity
+        const Eigen::VectorXd pt1 = geo_child->Global(testpoint);
+        // Map test point through relative geometry and then through geometry of
+        // parent
+        const Eigen::VectorXd tmp = geo_in_parent->Global(testpoint);
+        const Eigen::VectorXd pt2 = geo_parent->Global(tmp);
+        // Check whether both mappings yields the same point
+        // Note a valid test, in case the entity is not an AFFINE image of
```
Comment (craffael) [inline lib/lf/refinement/test/refinement_test_utils.cc]: "Not a valid test" ;)

**Comment by craffael** [comment]
@hiptmair What do you think if we implement also the `ParentEntity()` function as described in [here](https://github.com/craffael/lehrfempp/pull/167#pullrequestreview-328601695). If you want can also to the implementation, I think it's trivial. The main advantage would be, that the user doesn't need to use the `ParentInfos()` method which is for internal use only (#165 ). I know it's detail, but it's also very simple to implement :)

**Comment by craffael** [comment]
Also the documentation for the `GeometryInParent()` method is still missing.

File: lib/lf/refinement/test/refinement_test_utils.cc (hunk 1 above)
Comment (hiptmair) [reply lib/lf/refinement/test/refinement_test_utils.cc]: This method has been implemented.

File: lib/lf/refinement/test/refinement_test_utils.cc (hunk 2 above)
Comment (hiptmair) [reply lib/lf/refinement/test/refinement_test_utils.cc]: Fixed

File: lib/lf/refinement/mesh_hierarchy.cc [hunk 3]
```diff
@@ -345,9 +347,15 @@ void MeshHierarchy::RefineMarked() {
...
   } while (!refinement_complete);
 
+  // Create finer mesh according to set refinment edges
   PerformRefinement();
+  // Finish initialization
+  initGeometryInParent();
 }  // end RefineMarked
 
+/*
+ * TODO: Initialization of geometry field of ParentInfo Structure
```
[the remark this answers is not in the document]
Comment (hiptmair) [reply lib/lf/refinement/mesh_hierarchy.cc]: Removed

**Pull request merged**